File: common/src/main/java/com/hartwig/actin/clinical/serialization/DrugJson.java
package com.hartwig.actin.clinical.serialization;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

import com.google.gson.reflect.TypeToken;
import com.hartwig.actin.clinical.datamodel.treatment.Drug;

import org.jetbrains.annotations.NotNull;

public class DrugJson {
    @NotNull
    public static List<Drug> read(@NotNull String file) throws IOException {
        String contents = Files.readString(Path.of(file));
        return ClinicalGsonDeserializer.create().fromJson(contents, new TypeToken<List<Drug>>() {
        }.getType());
    }
}

File: clinical/src/main/kotlin/com/hartwig/actin/clinical/serialization/DrugJson.kt
package com.hartwig.actin.clinical.serialization

import com.google.gson.reflect.TypeToken
import com.hartwig.actin.clinical.datamodel.treatment.Drug
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

object DrugJson {
    @Throws(IOException::class)
    fun read(file: String): List<Drug> {
        val contents = Files.readString(Path.of(file))
        return ClinicalGsonDeserializer.create().fromJson(contents, object : TypeToken<List<Drug>>() {}.type)
    }
}